### DISCUSSION TITLE
use GATv2Conv or GATConv, but RuntimeError: Not compiled with CUDA support occured

### DISCUSSION THREAD

**Discussion by yayuanzi8** [pyg-team/pytorch_geometric#3642] [Q&A] [answered]
The error message is :

>   File "/home/cxj/anaconda3/envs/code_graph_nn_pyg/lib/python3.7/site-packages/torch_geometric/nn/conv/gatv2_conv.py", line 156, in forward
    out = self.propagate(edge_index, x=(x_l, x_r), size=size)
  File "/home/cxj/anaconda3/envs/code_graph_nn_pyg/lib/python3.7/site-packages/torch_geometric/nn/conv/message_passing.py", line 309, in propagate
    out = self.message(**msg_kwargs)
  File "/home/cxj/anaconda3/envs/code_graph_nn_pyg/lib/python3.7/site-packages/torch_geometric/nn/conv/gatv2_conv.py", line 183, in message
    alpha = softmax(alpha, index, ptr, size_i)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "/home/cxj/anaconda3/envs/code_graph_nn_pyg/lib/python3.7/site-packages/torch_scatter/scatter.py", line 160, in softmax
        return scatter_min(src, index, dim, out, dim_size)[0]
    elif reduce == 'max':
        return scatter_max(src, index, dim, out, dim_size)[0]
               ~~~~~~~~~~~ <--- HERE
    else:
        raise ValueError
  File "/home/cxj/anaconda3/envs/code_graph_nn_pyg/lib/python3.7/site-packages/torch_scatter/scatter.py", line 72, in scatter_max
        out: Optional[torch.Tensor] = None,
        dim_size: Optional[int] = None) -> Tuple[torch.Tensor, torch.Tensor]:
    return torch.ops.torch_scatter.scatter_max(src, index, dim, out, dim_size)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
RuntimeError: Not compiled with CUDA support

Version info：

> torch                 1.8.1
torch-cluster         1.5.9
torch-geometric       2.0.2
torch-scatter         2.0.9
torch-sparse          0.6.12
torch-spline-conv     1.2.1

cuda version is 10.1

**Comment by rusty1s** [comment, marked as answer]
Can you say something about how you did install `torch-scatter`? It looks like it only did install the CPU version in your case.

**Reply by yayuanzi8** [reply]
> Can you say something about how you did install `torch-scatter`? It looks like it only did install the CPU version in your case.

I ran pip install torch-scatter torch-sparse torch-cluster torch-spline-conv torch-geometric -f https://data.pyg.org/whl/torch-1.8.1+cu101.html

**Reply by yayuanzi8** [reply]
I can use scatter_mean, scatter_add using GPU, but only can't run scatter_max

**Reply by rusty1s** [reply]
Yes, this is to be expected as we use `torch.scatter_add_` internally for these variants. Can you try to uninstall `torch-scatter` and post the installation log of:
```
pip uninstall torch-scatter
pip install --no-cache-dir --verbose --no-index torch-scatter -f https://data.pyg.org/whl/torch-1.8.1+cu101.html
```

**Reply by yayuanzi8** [reply]
After reinstall torch-scatter by running `pip install --no-cache-dir --verbose --no-index torch-scatter -f https://data.pyg.org/whl/torch-1.8.1+cu101.html`，the problem solved，now I can use GATConv and GATv2Conv. The log is:

> Using pip 21.2.2 from /home/cxj/anaconda3/envs/code_graph_nn_pyg/lib/python3.7/site-packages/pip (python 3.7)
Looking in links: https://data.pyg.org/whl/torch-1.8.1+cu101.html
Collecting torch-scatter
  Downloading https://data.pyg.org/whl/torch-1.8.0%2Bcu101/torch_scatter-2.0.8-cp37-cp37m-linux_x86_64.whl (7.9 MB)
     |████████████████████████████████| 7.9 MB 870 kB/s 
Installing collected packages: torch-scatter
Successfully installed torch-scatter-2.0.8

I found the version is 2.0.8, instead of 2.0.9. But why？

**Reply by rusty1s** [reply]
Ah, you were using an older PyTorch version. For PyTorch 1.8.1, binaries of `torch-scatter` do only exist till version `2.0.8`, see [here](https://data.pyg.org/whl/torch-1.8.1+cu101.html). This shouldn't be a problem at all.

**Reply by yayuanzi8** [reply]
Thank you for your help.
